src/main/java/core/basesyntax/service/implementation/AddOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.implementation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.dto.FruitRecordDto;
+import core.basesyntax.service.FruitOperationHandler;
+
+public class AddOperation implements FruitOperationHandler {
+    @Override
+    public int apply(FruitRecordDto fruitRecordDto) {
+        String fruitName = fruitRecordDto.getFruitName();
+        int quantity = fruitRecordDto.getQuantity();
+        if (quantity < 0) {
+            throw new RuntimeException("Incorrect input value: " + quantity);
+        }
+        int oldQuantity = Storage.fruitsContainer.get(fruitName);
```

Comment:
you can get here NPE because get() in map may return null. Better to use avoid this case by using optional